File: scripts/check-event.py
import urllib.request, os

env = {}
with open('.env.local', 'r') as f:
    for line in f:
        if "=" in line and not line.startswith('#'):
            k, v = line.strip().split('=', 1)
            env[k] = v

host = env['VITE_SUPABASE_URL']
key = env['SUPABASE_SERVICE_ROLE_KEY']

req = urllib.request.Request(f'{host}/rest/v1/events?id=eq.d4163530-f5e1-4556-b27b-bc60e35562e7&select=title,starts_at,venue,image_url', headers={'apikey': key, 'Authorization': 'Bearer ' + key, 'Accept': 'application/vnd.pgrst.object+json'})
try:
  res = urllib.request.urlopen(req)
  print('EVENT:', res.read().decode())
except urllib.error.HTTPError as e:
  print('ERROR:', e.read().decode())
except Exception as e:
  print('ERROR:', str(e))

File: scripts/check-event.js
import fs from 'fs';

const envContent = fs.readFileSync('.env.local', 'utf-8');
const env = {};
envContent.split('\n').forEach(line => {
    const match = line.match(/^([^=]+)=(.*)$/);
    if (match) {
        env[match[1].trim()] = match[2].trim().replace(/\r$/, '');
    }
});

const SUPABASE_URL = env['VITE_SUPABASE_URL'] || '';
const SUPABASE_SERVICE_KEY = env['SUPABASE_SERVICE_ROLE_KEY'] || '';

async function checkData() {
    console.log("Fetching event with ID: d4163530-f5e1-4556-b27b-bc60e35562e7");

    const eRes = await fetch(`${SUPABASE_URL}/rest/v1/events?id=eq.d4163530-f5e1-4556-b27b-bc60e35562e7&select=title,starts_at,venue,image_url`, {
        headers: {
            'apikey': SUPABASE_SERVICE_KEY,
            'Authorization': `Bearer ${SUPABASE_SERVICE_KEY}`,
            'Accept': 'application/vnd.pgrst.object+json' // single() behavior
        }
    });

    const eventOutput = await eRes.text();
    console.log("Event Output:", eventOutput);
}

checkData();
